connectors/class-connector-jetpack.php

```diff
@@ -373,15 +373,18 @@ public function callback_jetpack_log_entry( array $entry ) {
 			}
 
 			$user       = new \WP_User( $user_id );
-			$user_email = $user->user_email;
-			$user_login = $user->user_login;
-			$context    = 'users';
-			$action     = $method;
-			$meta       = compact( 'user_id', 'user_email', 'user_login' );
-			$message    = sprintf(
-				/* translators: %1$s: a user display name, %2$s: a status and the connection either "from" or "to" (e.g. "Jane Doe", "unlinked from") */
+			$user_email = ! empty( $user->user_email ) ? $user->user_email : '';
+			$user_login = ! empty( $user->user_login ) ? $user->user_login : '';
+
+			/* translators: %d is the user ID */
+			$user_display_name = ! empty( $user->display_name ) ? $user->display_name : sprintf( __( 'Unknown user %d', 'stream' ), $user_id );
+			$context           = 'users';
+			$action            = $method;
+			$meta              = compact( 'user_id', 'user_email', 'user_login' );
+			$message           = sprintf(
+			/* translators: %1$s: a user display name, %2$s: a status and the connection either "from" or "to" (e.g. "Jane Doe", "unlinked from") */
 				__( '%1$s\'s account %2$s Jetpack', 'stream' ),
-				$user->display_name,
+				esc_html( $user_display_name ),
```

Comment:
<!-- general note -->
**📝 note:** Actually, I'm a bit confused if this should or should not be escaped here already. Other strings (unchanged by this PR) are sometimes escaped and sometimes not escaped 🤷

Reply:
So it looks like all record fields, including the message/summary, go through a sanitizer before they are stored in the DB. All HTML tags are stripped by the sanitizer:

https://github.com/xwp/stream/blob/275ed1d55078dce525d991f7cb4d99479249a694/classes/class-db.php#L118-L128

Since we don't expect any HTML tags in the user display name (and we definitely don't want any tag to be displayed there), we won't escape HTML letting `wp_strip_all_tags` do its job.

But I guess in some cases we actually want to escape HTML so that tags are not stripped out and instead presented in the logs for better understanding of what has happened.